src/main/java/frc/robot/commands/intake/IntakeAndIndex.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot.commands.intake;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.ScheduleCommand;
+import frc.robot.subsystems.IndexerSubsystem;
+import frc.robot.subsystems.IntakeSubsystem;
+import friarLib2.commands.RunUntilCommand;
+
+/**
+ * While this command is running, run the intake. When this command 
+ * starts, tell the indexer to index a cargo.
+ */
+public class IntakeAndIndex extends ParallelCommandGroup {
+
+    private Timer timer = new Timer();
+
+    public IntakeAndIndex (IntakeSubsystem intake, IndexerSubsystem indexer) {
+        timer.reset();
+        timer.stop();
+        
+        addCommands(
+            new ScheduleCommand( // We schedule the indexer's command separately so that it can run after this one terminates
+                new RunUntilCommand(
+                    new IndexOneCargo(indexer), // This will run either until it finishes or until five seconds have elapsed after the intake retracts
+                    () -> timer.get() >= 5 // The timer only starts running after the intake retracts
```

Comment:
This might be solving the infinite loop issue I saw above, if so then this would do better inside of the index one cargo function since the timer is integral to the system not breaking if this function is called and it fails somehow. 

Ex: Someone else is editing code and required the index one cargo as part of some new Auto they are creating. without the previous knowledge that you have on this function, they would not know they needed a timer to stop the index one cargo in the event of a failure. This could lead to an infinite issue that could paralyze the intake for some or all of the match.

Reply:
As long as a command is scheduled as interruptible (they are by default), then scheduling another command that requires the indexer will be able to regain control over it.